semver.py

```diff
@@ -47,10 +28,44 @@ def cmp(a, b):
         return (a > b) - (a < b)
 
 
+def deprecated(replace=None):
+    """
+    Decorates a function to output a deprecation warning.
+
+    This function will be removed once major version 3 of semver is
+    released.
+    """
+    # we can't use the 'nonlocal' keyword in Python2, so we need
+    # to circumvent that with a dictionary:
+    r = {"r": replace}
+
+    def decorator(func):
+        @wraps(func)
+        def wrapper(*args, **kwargs):
+            # nonlocal replace
+            replace = r["r"] if r["r"] is not None else func.__name__
+            msg = (
+                "Function 'semver.{f}' is deprecated. "
+                "Use the respective 'semver.VersionInfo.{r}' instead."
+            )
+            warnings.warn(
+                msg.format(f=func.__name__, r=replace), category=DeprecationWarning
+            )
+            return func(*args, **kwargs)
+
+        return wrapper
+
+    return decorator
+
+
+@deprecated()
 def parse(version):
     """
     Parse version to major, minor, patch, pre-release, build parts.
 
+    .. deprecated:: 2.9.2
+       Use :func:`semver.VersionInfo.format_parse` instead.
```

Comment:
There's no `format_parse`.

Reply:
Right, that was a copy-and-paste error. Fixed.